network-api/networkapi/campaign/views.py

```diff
@@ -0,0 +1,85 @@
+from rest_framework.decorators import api_view, parser_classes, permission_classes
+from rest_framework.response import Response
+from rest_framework.parsers import JSONParser
+from rest_framework import status, permissions
+from django.core.exceptions import ObjectDoesNotExist
+from django.conf import settings
+from datetime import datetime
+import boto3
+import logging
+import json
+
+from networkapi.campaign.models import Petition
+
+if settings.AWS_SQS_ACCESS_KEY_ID:
+    sqs = boto3.client(
+        'sqs',
+        region_name=settings.AWS_SQS_REGION,
+        aws_access_key_id=settings.AWS_SQS_ACCESS_KEY_ID,
+        aws_secret_access_key=settings.AWS_SQS_SECRET_ACCESS_KEY,
+    )
+queue_url = settings.PETITION_SQS_QUEUE_URL
+logger = logging.getLogger(__name__)
+
+
+@api_view(['POST'])
+@parser_classes((JSONParser,))
+@permission_classes((permissions.AllowAny,))
+def petition_submission_view(request, pk):
+    try:
+        petition = Petition.objects.get(id=pk)
+    except ObjectDoesNotExist:
+        return Response(
+            {'error': 'Invalid petition id'},
+            status=status.HTTP_400_BAD_REQUEST,
+        )
+
+    data = {
+        'entry.1017670790': request.data['givenNames'],
```

Comment:
@gideonthomas This will need to be updated when we want to do this for other petitions, right?

Reply:
If this is going to be a long-ish term solution - i.e. other petitions, we need to abstract these field key names to  fields in the petition model.